src/charm.py

```diff
@@ -1122,6 +1140,23 @@ def _on_debug_ssh_relation_changed(self, _: ops.RelationChangedEvent) -> None:
             state.runner_config.virtual_machine_resources,
         )
 
+    @catch_charm_errors
+    def _on_image_relation_changed(self, _: ops.RelationChangedEvent) -> None:
+        """Handle debug ssh relation changed event."""
+        state = self._setup_state()
+
+        if state.instance_type != InstanceType.OPENSTACK:
+            return
```

Comment:
Should we go into blocked state to inform the user that they cannot use this relationship with lxd?

Reply:
I think it's okay - the LXD runners should build images itself. Going it to BlockedState suggests that the relation should be removed for the runners to work.
How about a warning log?

Reply:
When the user relates to the image relation, they have a certain expectation and that expectation is not met with LXD. 
I think BlockedState might be appropriate as it is very visible in the juju status output. An alternative would be to set a status message for the Active state.

The logs are usually not read if everything seems to be working fine.

I think BlockedState could also be used if the core functionality is still working, https://juju.is/docs/sdk/status#heading--possible-statuses leaves room for interpretation:

>Blocked: the device requires manual intervention by the Juju user, as specified by the charm author (see status notes or charm docs).

@gregory-schiano what do you think should happen in such cases?

Reply:
Perhaps a message on the current status would be fit.
```
StatusBase.from_name(self.unit.status.name, "LXD runners do not require image relation to be set.")
```
What do you think?

Reply:
So the workload will still work properly but the relation is "useless"
I think a message in the active status is fine, but we should make sure the message is persistent so probably use a collect status event and put the logic there.
Beware that using collect status can override any status set before, so it has consequences on how you manages it

Reply:
Will add collect status on a follow up PR, since it adds overhead of a review.